staging/src/k8s.io/api/scheduling/v1alpha1/types.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient=true
+// +nonNamespaced=true
+
+// PriorityClass defines mapping from a priority class name to the priority
+// integer value. The value can be any valid integer.
+type PriorityClass struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// The value of this priority class. This is the actual priority that pods
+	// receive when they have the name of this class in their pod spec.
+	Value int32 `json:"value" protobuf:"bytes,2,opt,name=value"`
+
+	// GlobalDefault specifies whether this PriorityClass should be considered as
+	// the default priority for pods that do not have any priority class and their
+	// namespace do not have any other default priority.
+	// +optional
+	GlobalDefault bool `json:"globalDefault,omitempty" protobuf:"bytes,3,opt,name=globalDefault"`
```

Comment:
Can you write more explicitly what the rules for defaulting are?

I'm going to assume PriorityClass can be used both at top-level and within a namespace (otherwise I'm totally confused).

It seems there are four scenarios
* a PriorityClass is top-level and marked GlobalDefault=false
* a PriorityClass is top-level and marked GlobalDefault=true
* a PriorityClass is within a namespace and marked GlobalDefault=false
* a PriorityClass is within a namespace and marked GlobalDefault=true

What is the rule for determining the PriorityClass that applies to a pod?

For example, one possibility would be
```
if (pod does not set PriorityClass)
    if (pod's namespace has a PriorityClass object marked GlobalDefault)
        use the corresponding priority
    else if (top-level has a PriorityClass object marked GlobalDefault)
         use the corresponding priority
    else
         // not sure what to do here 
else // pod sets PriorityClass
    if (pod refers to PriorityClass that exists within its namespace)
         use the corresponding priority
    else if (pod refers to PriorityClass that exists at top-level)
          use the corresponding priority
    else
       // not sure what to do here 
```

Also, I thought we were going to use an annotation for marking default, to match the semantics of StorageClass (or whichever other object had the idea of marking one as default)? We should avoid introducing a new way to specify default, if possible.